src/components/lib/FlowGrid/background-container.js

```diff
@@ -20,23 +22,36 @@ export class BackgroundContainer extends Component {
   displayName: 'BackgroundContainer'
 
   static propTypes = {
-    rowHeights: PropTypes.array.isRequired,
+    rowCount: PropTypes.number.isRequired,
     edges: PropTypes.array.isRequired,
     selectedRegion: PTRegion,
     copiedRegion: PTRegion,
   }
 
+  state = {
+    rowHeights: []
+  }
+
+  componentWillMount() {
+    this.setState({rowHeights: _.map(upto(this.props.rowCount), (r, i) => this.props.getRowHeight(i))})
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({rowHeights: _.map(upto(nextProps.rowCount), (r, i) => nextProps.getRowHeight(i))})
+  }
+
   shouldComponentUpdate(nextProps, nextState) {
     return (
       !_.isEqual(this.props.copiedRegion, nextProps.copiedRegion) ||
       !_.isEqual(this.props.selectedRegion, nextProps.selectedRegion) ||
       !_.isEqual(this.props.edges, nextProps.edges) ||
-      !_.isEqual(this.props.rowHeights, nextProps.rowHeights)
+      !_.isEqual(this.state.rowHeights, nextState.rowHeights)
```

Comment:
this seems sketchy to me.  I'm not sure if _.isEqual works for arrays, though we are using it for arrays and objects above.